### PR TITLE
Plugin Modal: Apply (save) button + normalize

### PR DESCRIPTION
**Pull request by Ruud68**
Pull Request for Issue # .

### Summary of Changes
This PR normalizes the plugin modal as currently used in com_redirect. It also adds a [save] button to the modal footer.

### Testing Instructions
1. normalization:
1.1 disable the plg_redirect
1.2 go to component / Redirect
1.3 in the error message click on the link: "Redirect System Plugin"
[A]
2. save button
2.1 change plugin status in modal to 'enabled'
2.2 change Collect URLs to 'disabled'
2.3 click on [Save] button (the modal should stay open)
2.4 click on [Close] button
[B] 

### Expected result
[A] The modal should open and should be same size / form / shape / etc. as other modals (like edit article in menu single article

[B] after closing the modal, the com_redirect page should refresh and the error message must be replaced with a Notice 'The Redirect Plugin is enabled. The 'Collect URLs' option in the Redirect System Plugin is disabled. Error page URLs will not be collected by this component.'
If you did not change the Collect URLs vanlue to 'disabled' there will be no warning / notice.

### Actual result
[A] modal is small (width and heigth) the other modals
[B] There is no [Save] button, just a [Close]  and [Save & Close] button

### Documentation Changes Required
None

### CODE REVIEW

File: administrator/components/com_plugins/views/plugin/tmpl/edit.php [hunk 1]
```diff
@@ -17,22 +17,33 @@
...
+$isModal  = $input->get('layout') == 'modal' ? true : false;
+$layout   = $isModal ? 'modal' : 'edit';
+$tmpl     = $isModal || $input->get('tmpl', '', 'cmd') === 'component' ? '&tmpl=component' : '';
+
 JFactory::getDocument()->addScriptDeclaration("
 	Joomla.submitbutton = function(task)
 	{
-		if (task == 'plugin.cancel' || document.formvalidator.isValid(document.getElementById('style-form'))) {
+		if (task == 'plugin.cancel' || document.formvalidator.isValid(document.getElementById('style-form')))
+		{
```
Comment (dgrammatiko) [inline administrator/components/com_plugins/views/plugin/tmpl/edit.php]: Please let the curly brace on the same line as the `if`, javascript doesn't have the same code style as PHP

File: administrator/components/com_plugins/views/plugin/tmpl/edit.php [hunk 2]
```diff
@@ -17,22 +17,33 @@
...
+// In case of modal
+$isModal  = $input->get('layout') == 'modal' ? true : false;
+$layout   = $isModal ? 'modal' : 'edit';
+$tmpl     = $isModal || $input->get('tmpl', '', 'cmd') === 'component' ? '&tmpl=component' : '';
+
 JFactory::getDocument()->addScriptDeclaration("
 	Joomla.submitbutton = function(task)
 	{
-		if (task == 'plugin.cancel' || document.formvalidator.isValid(document.getElementById('style-form'))) {
+		if (task == 'plugin.cancel' || document.formvalidator.isValid(document.getElementById('style-form')))
```
Comment (C-Lodder) [inline administrator/components/com_plugins/views/plugin/tmpl/edit.php]: `==` >> `===`

File: administrator/components/com_plugins/views/plugin/tmpl/edit.php [hunk 3]
```diff
@@ -17,22 +17,33 @@
 JHtml::_('bootstrap.tooltip');
 $this->fieldsets = $this->form->getFieldsets('params');
 
+$input = JFactory::getApplication()->input;
+
+// In case of modal
+$isModal  = $input->get('layout') == 'modal' ? true : false;
```
Comment (C-Lodder) [inline administrator/components/com_plugins/views/plugin/tmpl/edit.php]: `==` >> `===`

**Comment by Ruud68** [comment]
Thanks @C-Lodder, changes made as pointed out by you :)

**Comment by ladyjer** [comment]
I have tested this item :white_check_mark: successfully on 95a6d3b9e203d7e6d9bf5fb97b97871d72a314a5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18913">issues.joomla.org/tracker/joomla-cms/18913</a>.</sub>

**Comment by Ruud68** [comment]
Thanks for testing @ladyjer :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18913">issues.joomla.org/tracker/joomla-cms/18913</a>.</sub>

File: administrator/components/com_plugins/views/plugin/tmpl/edit.php [hunk 4]
```diff
@@ -17,22 +17,30 @@
...
 			Joomla.submitform(task, document.getElementById('style-form'));
-		}
-		
-		if (self !== top) {
-			window.top.setTimeout('window.parent.location = window.top.location.href', 1000);
-			window.parent.jQuery('#plugin" . $this->item->extension_id . "Modal').modal('hide');
+
+			if (task !== 'plugin.apply') {
+				if (self !== top ) {
+                        window.top.setTimeout('window.parent.location = window.top.location.href', 1000);
```
Comment (Quy) [inline administrator/components/com_plugins/views/plugin/tmpl/edit.php]: Lines 34-36: Fix whitespace. Use tabs.

**Comment by infograf768** [comment]
Test fine here, waiting for cs to mark it such.

**Comment by Ruud68** [comment]
@Quy @infograf768 Fixed, not sure why my own code style checker didn't 'catch' these. Thanks for reporting and testing!

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on cc34df85d886b200e5cb64886f9cea9d88558e2b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18913">issues.joomla.org/tracker/joomla-cms/18913</a>.</sub>

**Comment by infograf768** [comment]
RTC. Thanks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18913">issues.joomla.org/tracker/joomla-cms/18913</a>.</sub>

**Pull request merged**